File: src/cleansia_ios/CleansiaPartner/Sources/Features/Orders/WorkContractCard.swift
import CleansiaCore
import SwiftUI

/// The cleaner's standing on the contract for work: the acceptance line with a way to read the
/// accepted text, or — for a seat an administrator placed — the prompt to accept before starting.
/// Nothing renders for a cleaner who is not on the crew; the take path carries its own contract.
struct WorkContractCard: View {
    @Environment(\.locale) private var locale
    let standing: WorkContractStanding
    var onAccept: () -> Void = {}
    var onRead: (String) -> Void = { _ in }

    var body: some View {
        switch standing {
        case .none:
            EmptyView()
        case .pending:
            PendingBanner(onAccept: onAccept)
                .id(locale.identifier)
        case let .accepted(acceptanceId, acceptedOn, documentVersion):
            AcceptedLine(
                acceptedOn: OrdersFormat.dateTime(acceptedOn, locale: locale),
                documentVersion: documentVersion,
                onRead: { onRead(acceptanceId) }
            )
            .id(locale.identifier)
        }
    }
}

/// The attention treatment the access card uses: tinted, inset, rounded — this is the one thing on
/// the detail the cleaner has to act on before the work.
private struct PendingBanner: View {
    let onAccept: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: Spacing.xs) {
            ContractTitle(ink: CleansiaColors.warningStar)
            Text(L10n.WorkContract.pendingBanner)
                .font(CleansiaTypography.bodyMedium)
                .foregroundColor(CleansiaColors.onSurface)
                .fixedSize(horizontal: false, vertical: true)
            CleansiaPrimaryButton(L10n.WorkContract.acceptCta, size: .medium, action: onAccept)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(Spacing.m)
        .background(CleansiaColors.warningStar.opacity(0.12), in: RoundedRectangle(cornerRadius: CornerRadius.medium))
    }
}

/// Flat like every other section on the sheet (`OrderSectionCard`): a fact, not a prompt.
private struct AcceptedLine: View {
    let acceptedOn: String
    let documentVersion: String
    let onRead: () -> Void

    var body: some View {
        VStack(alignment: .leading, spacing: Spacing.xs) {
            ContractTitle(ink: CleansiaColors.onSurface)
            Text(L10n.WorkContract.acceptedLine(acceptedOn, documentVersion))
                .font(CleansiaTypography.bodyMedium)
                .foregroundColor(CleansiaColors.onSurface)
                .fixedSize(horizontal: false, vertical: true)
            CleansiaTextLink(L10n.WorkContract.read, action: onRead)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(.vertical, Spacing.m)
        .background(CleansiaColors.surface)
    }
}

private struct ContractTitle: View {
    let ink: Color

    var body: some View {
        Label(L10n.WorkContract.title, systemImage: "doc.text")
            .font(CleansiaTypography.titleMedium)
            .foregroundColor(ink)
    }
}

#if DEBUG
    struct WorkContractCard_Previews: PreviewProvider {
        static var previews: some View {
            VStack(spacing: Spacing.m) {
                WorkContractCard(standing: .pending)
                WorkContractCard(standing: .accepted(
                    acceptanceId: "acc-1",
                    acceptedOn: Date(timeIntervalSince1970: 1_786_200_000),
                    documentVersion: "2026-09-20"
                ))
            }
            .padding()
            .background(CleansiaColors.surface)
        }
    }
#endif

File: src/cleansia_android/partner-app/src/main/java/cz/cleansia/partner/features/orders/WorkContractCard.kt
package cz.cleansia.partner.features.orders

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Description
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import cz.cleansia.core.format.formatOrderDateTime
import cz.cleansia.core.ui.components.CleansiaPrimaryButton
import cz.cleansia.core.ui.components.CleansiaTextLink
import cz.cleansia.core.ui.theme.Spacing
import cz.cleansia.partner.R

/**
 * The cleaner's standing on the contract for work: the acceptance line with a way to read the
 * accepted text, or — for a seat an administrator placed — the prompt to accept before starting.
 * Nothing renders for a cleaner who is not on the crew; the take path carries its own contract.
 */
@Composable
fun WorkContractCard(
    standing: WorkContractStanding,
    onAccept: () -> Unit,
    onRead: (acceptanceId: String) -> Unit,
    modifier: Modifier = Modifier,
) {
    when (standing) {
        WorkContractStanding.None -> Unit
        WorkContractStanding.Pending -> ContractSurface(
            container = MaterialTheme.colorScheme.tertiaryContainer,
            content = MaterialTheme.colorScheme.onTertiaryContainer,
            modifier = modifier,
        ) {
            Text(
                text = stringResource(R.string.work_contract_pending_banner),
                style = MaterialTheme.typography.bodyMedium.copy(fontWeight = FontWeight.SemiBold),
                color = MaterialTheme.colorScheme.onTertiaryContainer,
            )
            Spacer(Modifier.height(Spacing.S))
            CleansiaPrimaryButton(
                text = stringResource(R.string.work_contract_accept_cta),
                onClick = onAccept,
            )
        }
        is WorkContractStanding.Accepted -> ContractSurface(
            container = MaterialTheme.colorScheme.surface,
            content = MaterialTheme.colorScheme.onSurfaceVariant,
            modifier = modifier,
        ) {
            Text(
                text = stringResource(
                    R.string.work_contract_accepted_line,
                    formatOrderDateTime(standing.acceptedOn),
                    standing.documentVersion,
                ),
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurface,
            )
            CleansiaTextLink(
                text = stringResource(R.string.work_contract_read),
                onClick = { onRead(standing.acceptanceId) },
            )
        }
    }
}

@Composable
private fun ContractSurface(
    container: androidx.compose.ui.graphics.Color,
    content: androidx.compose.ui.graphics.Color,
    modifier: Modifier = Modifier,
    body: @Composable () -> Unit,
) {
    Surface(
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(16.dp),
        color = container,
        tonalElevation = 0.dp,
    ) {
        Column(modifier = Modifier.padding(vertical = Spacing.M)) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(Spacing.XS),
            ) {
                Icon(
                    imageVector = Icons.Outlined.Description,
                    contentDescription = null,
                    modifier = Modifier.size(18.dp),
                    tint = content,
                )
                Text(
                    text = stringResource(R.string.work_contract_title),
                    style = MaterialTheme.typography.titleSmall.copy(fontWeight = FontWeight.SemiBold),
                    color = content,
                )
            }
            Spacer(Modifier.height(Spacing.XS))
            body()
        }
    }
}
